maup/repair.py

```diff
@@ -1,27 +1,1217 @@
+                    ):  # Less than 0.01%
+                        component_num_list.remove(
+                            c_ind
+                        )  # Tells us to take out this component later
+                        possible_intersect_integer_indices = [
+                            *set(
+                                numpy.ndarray.flatten(
+                                    spatial_index.query(this_fragment)
+                                )
+                            )
+                        ]
+                        possible_intersect_indices = [
+                            (index_by_iloc[k])
+                            for k in possible_intersect_integer_indices
+                        ]
+
+                        shared_perimeters = []
+                        for g_ind2 in possible_intersect_indices:
+                            if (
+                                g_ind2 != g_ind
+                                and not (this_fragment.boundary)
+                                .intersection(
+                                    reconstructed_df["geometry"][g_ind2].boundary
+                                )
+                                .is_empty
+                            ):
+                                shared_perimeters.append(
+                                    (
+                                        g_ind2,
+                                        (this_fragment.boundary)
+                                        .intersection(
+                                            reconstructed_df["geometry"][
+                                                g_ind2
+                                            ].boundary
+                                        )
+                                        .length,
+                                    )
+                                )
+
+                        max_shared_perim = sorted(
+                            shared_perimeters, key=lambda tup: tup[1]
+                        )[-1]
+                        poly_to_add_to = max_shared_perim[0]
+                        reconstructed_df["geometry"][poly_to_add_to] = unary_union(
+                            [
+                                reconstructed_df["geometry"][poly_to_add_to],
+                                this_fragment,
+                            ]
+                        )
+
+                if len(component_num_list) == 1:
+                    reconstructed_df["geometry"][g_ind] = reconstructed_df["geometry"][
+                        g_ind
+                    ].geoms[component_num_list[0]]
+                elif len(component_num_list) > 1:
+                    reconstructed_df["geometry"][g_ind] = MultiPolygon(
+                        [
+                            reconstructed_df["geometry"][g_ind].geoms[c_ind]
+                            for c_ind in component_num_list
+                        ]
+                    )
+                else:
+                    print(
```

Comment:
Ideally, this should show up in a `Report` object (exact name and structure TBD) rather than just being logged. This would make it easier for users to, say, loop over the indices of badly disconnected geometries.